### PR TITLE
Update due to changes in tesseract_visualization interface

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This update to the latest visualization interface in Tesseract. It also switches the default for descartes to use floats.

### CODE REVIEW

File: tesseract_motion_planners/src/core/utils.cpp [hunk 1]
```diff
@@ -40,6 +40,150 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
...
+          poses.push_back(*cwp);
+        }
+        else
+        {
+          poses.push_back(state->link_transforms.at(manip_info.working_frame) * (*cwp));
+        }
+      }
+      else
+      {
+        CONSOLE_BRIDGE_logError("toToolpath: Unsupported Waypoint Type!");
```
Comment (marip8) [inline tesseract_motion_planners/src/core/utils.cpp:124]: Would we want a hard failure (exception) here instead? Or is this function primarily intended for visualization purposes?

File: tesseract_motion_planners/src/core/utils.cpp [hunk 2]
```diff
@@ -40,6 +40,150 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
...
 {
+tesseract_common::Toolpath toToolpath(const Instruction& instruction,
+                                      const tesseract_environment::Environment::ConstPtr& env)
+{
+  using namespace tesseract_planning;
+
+  if (env == nullptr)
+    throw std::runtime_error("toToolpath: The environment is a nullptr!");
+
+  tesseract_common::Toolpath toolpath;
```
Comment (marip8) [inline tesseract_motion_planners/src/core/utils.cpp:51]: Out of curiosity, does a vector containing a vector of Eigen objects also need the special aligned allocator? I was under the impression that only the inner vector needs it

**Review by marip8 — COMMENTED** [review]
I think this looks good overall; I just left some general comments. You might consider splitting the Descartes changes out into a separate PR for clarity since it addresses something totally different

File: tesseract_motion_planners/src/core/utils.cpp [hunk 3]
```diff
@@ -40,6 +40,150 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
...
+      }
+      else
+      {
+        poses.push_back(state->link_transforms.at(manip_info.working_frame) * (*cwp));
+      }
+    }
+    else
+    {
+      CONSOLE_BRIDGE_logError("toToolpath: Unsupported Waypoint Type!");
+    }
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_motion_planners/src/core/utils.cpp:176]: I broke it out into a separate function.

File: tesseract_motion_planners/src/core/utils.cpp (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract_motion_planners/src/core/utils.cpp:124]: I think an exception is good.

File: tesseract_motion_planners/src/core/utils.cpp (hunk 2 above)
Comment (Levi-Armstrong) [reply tesseract_motion_planners/src/core/utils.cpp:51]: Yea anything that contains an Eigen aligned object must also use the aligned operator. Same reason classes that include member objects that require the Eigen alignment then you also have to have it in your class. It is infections.

**Comment by Levi-Armstrong** [comment]
@marip8 CI passed. You good with the changes?

**Comment by Levi-Armstrong** [comment]
I will clean up the commit history before merging.

**Review by marip8 — APPROVED** [review]

**Comment by mpowelson** [comment]
@Levi-Armstrong Care to comment for posterity why you changed the default to float for descartes? It sounds like a reasonable thing to do, but I don't see the reasoning mentioned.

**Comment by mpowelson** [comment]
@Levi-Armstrong Care to comment for posterity why you changed the default to float for descartes? It sounds like a reasonable thing to do, but I don't see the reasoning mentioned.

**Comment by Levi-Armstrong** [comment]
In the case of Tesseract all of the calculations are still done using double precision and Descartes is just storing the joint solution and cost using floating point which does not seem to change the accuracy of the results with reason. I think we could event get away with using Eigen::Half.

**Pull request merged**